src/content/docs/merge-queue/monorepo.mdx

```diff
@@ -0,0 +1,259 @@
+---
+title: Monorepo
+description: Optimize merge queue batching for monorepos with scopes.
+---
+
+In monorepo environments, not every pull request affects the entire codebase. Running all tests for
+every change wastes time and CI resources. Mergify's **scopes** feature allows you to intelligently
+batch pull requests based on which parts of your codebase they modify, dramatically improving merge
+queue efficiency.
+
+## Understanding Scopes
+
+Scopes define discrete areas of your monorepo (like packages, services, or components). When a pull
+request is created, Mergify automatically determines which scopes are affected and uses this
+information to optimize batching.
+
+Scopes can be determined in several ways depending on your project's needs:
+
+- **File patterns**: Match file paths to identify affected scopes (currently supported)
+
+- **Build system integration**: Future support for tools like Bazel, Nx, Turborepo, and others that
+  can provide scope information based on dependency graphs and build targets
+
+This flexibility allows you to use the approach that best fits your monorepo's architecture and
+existing tooling.
+
+### The Batching Challenge in Monorepos
+
+Without scopes, Mergify batches pull requests together regardless of what they change. This means:
+
+- A Python service change and a JavaScript frontend change might be batched together
+- Both sets of tests run even though they're completely independent
+- If one fails, both PRs are affected by the batch split process
+
+With scopes, Mergify can:
+
+- Batch together PRs that affect the same scopes (e.g., multiple Python changes)
+- Keep independent changes in separate batches
+- Reduce unnecessary CI runs for unrelated parts of your codebase
+
+## Configuring Scopes
+
+Define scopes in your `.mergify.yml` configuration file using file patterns:
+
+```yaml
+scopes:
+  source:
+    files:
+      python-api:
+        includes:
+          - api/**/*.py
+          - libs/shared/**/*.py
+      frontend:
+        includes:
+          - web/**/*.js
+          - web/**/*.jsx
+          - web/**/*.ts
+          - web/**/*.tsx
+      docs:
+        includes:
+          - docs/**/*.md
+          - docs/**/*.mdx
+
+queue_rules:
+  - name: default
+    batch_size: 5
+```
+
+In this example:
+- Changes to Python files in `api/` or `libs/shared/` get the `python-api` scope
+- Changes to frontend files in `web/` get the `frontend` scope
+- Documentation changes get the `docs` scope
+
+:::tip
+Mergify will intelligently batch PRs with overlapping scopes together. For example, if PR1 affects
+`python-api` and PR2 affects both `python-api` and `frontend`, they'll be batched together since
+they share a common scope.
+:::
+
+## Setting Up CI with Scopes
+
+To leverage scopes in your CI workflow, use the
+[gha-mergify-ci](https://github.com/Mergifyio/gha-mergify-ci) GitHub Action. This action detects
```

Comment:
I think we have to improve the readme of this repository to avoid any confusion

Reply:
I have a plan for that :)